configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -358,27 +375,19 @@ void activate(RemoteSession session) {
         long sessionId = session.getSessionId();
         Curator.CompletionWaiter waiter = createSessionZooKeeperClient(sessionId).getActiveWaiter();
         log.log(Level.FINE, () -> session.logPre() + "Getting session from repo: " + sessionId);
-        ApplicationSet app = session.ensureApplicationLoaded();
+        ApplicationSet app = ensureApplicationLoaded(session);
         log.log(Level.FINE, () -> session.logPre() + "Reloading config for " + sessionId);
         applicationRepo.reloadConfig(app);
         log.log(Level.FINE, () -> session.logPre() + "Notifying " + waiter);
-        session.notifyCompletion(waiter);
+        notifyCompletion(waiter, session);
         log.log(Level.INFO, session.logPre() + "Session activated: " + sessionId);
     }
 
-    public void deactivate(RemoteSession remoteSession) {
-        remoteSession.deactivate();
-    }
-
-    public void delete(RemoteSession remoteSession, Optional<LocalSession> localSession) {
+    public void deleteSession(RemoteSession remoteSession, Optional<LocalSession> localSession) {
         localSession.ifPresent(this::deleteLocalSession);
         remoteSession.deactivate();
```

Comment:
Shouldn't this also be `deactivate(remoteSession)`? This does nothing now.

Reply:
Yes, thanks